docs/architecture/adr-061-core-module-api.md

```diff
@@ -0,0 +1,490 @@
+type of store they need in their dependency injection (or manual) constructors.
+
+#### Event Service
+
+The event `Service` and `Manager` will be defined in the `cosmossdk.io/core/event` package.
+
+The event `Service` gives modules access to an event `Manager` which allows modules to emit typed and legacy,
+untyped events:
+
+```go
+package event
+type Service interface {
+    GetManager(context.Context) Manager
+}
+
+type Manager interface {
+    Emit(proto.Message) error
+    EmitLegacy(eventType string, attrs ...LegacyEventAttribute) error
+}
+```
+
+By definition, typed events emitted with `Emit` are part of consensus and can be observed by other modules while
+legacy events emitted by `EmitLegacy` are not considered to be part of consensus and cannot be observed by other
+modules.
+
+Design questions:
+* should we allow overriding event managers (like `sdk.Context.WithEventManager`)?
+* should there be a way to "silently" emit typed events that aren't part of consensus?
+
+#### Block Info Service
+
+The block info `Service` will be defined in the `cosmossdk.io/core/blockinfo` package.
+
+The `BlockInfo` service allows modules to get a limited set of information about the currently executing block
+that does not depend on any specific version of Tendermint:
+```go
+package blockinfo
+
+type Service interface {
+    GetBlockInfo(ctx context.Context) BlockInfo
+}
+
+type BlockInfo interface {
+    ChainID() string
+    Height() int64
+    Time() *timestamppb.Timestamp
+    Hash() []byte
+}
+```
+
+Only a limited set of modules need any other information from the Tendermint block header and specific versions
+of runtime modules tied to specific Tendermint versions can expose this functionality to modules that need it.
+The basic `BlockInfo` service is left fairly generic to insulate the vast majority of other modules from unneeded
+exposure to changes in the Tendermint protocol.
+
+#### Gas Service
+
+The gas `Service` and `Meter` types will be defined in the `cosmossdk.io/core/gas` package.
+
+The gas service handles tracking of gas consumptions at the block and transaction level and also allows
+for overriding the gas meter passed to child calls:
+```go
+package gas
+
+type Service interface {
+	GetMeter(context.Context) Meter
+	GetBlockMeter(context.Context) Meter
+	WithMeter(ctx context.Context, meter Meter) context.Context
+	WithBlockMeter(ctx context.Context, meter Meter) context.Context
+}
+```
+
+#### `grpc.ClientConnInterface`
+
+Runtime module implementations should provide an instance of `grpc.ClientConnInterface` as core service. This service
+will allow modules to send messages to and make queries against other modules as described in [ADR 033](./adr-033-protobuf-inter-module-comm.md).
+
+A single `grpc.ClientConnInterface` will be used for both MsgClient's and QueryClient's and the state machine will
+make the presence or absence of the protobuf option `cosmos.msg.v1.service` which will be required for all
```

Comment:
>and the state machine will make the presence or absence of the protobuf option cosmos.msg.v1.service which will be required for all valid Msg service types.

I don't understand this sentence, is there a typo or something?

Reply:
Yes, I'll fix